packages/@n8n/nodes-langchain/nodes/mcp/McpTrigger/McpServer.ts

```diff
@@ -217,13 +220,13 @@ export class McpServerSingleton {
 
 	private _serverData: McpServer;
 
-	private constructor(logger: Logger) {
-		this._serverData = new McpServer(logger);
+	private constructor(serverName: string, logger: Logger) {
+		this._serverData = new McpServer(serverName, logger);
 	}
 
-	static instance(logger: Logger): McpServer {
+	static instance(serverName: string, logger: Logger): McpServer {
 		if (!McpServerSingleton.#instance) {
-			McpServerSingleton.#instance = new McpServerSingleton(logger);
+			McpServerSingleton.#instance = new McpServerSingleton(serverName, logger);
```

Comment:
won't this create a instance with a fixed name that all subsequent mcp servers will end up then reusing?
do we even need the singleton? it looks like it'll cause more trouble than it's worth.

Reply:
Thinking the same, I marked it as a draft PR till we decide. Here are some things to be considered:
* I think there needs to be only one instance/per MCP trigger node.
* That instance can be re-instantiated when the tools / name change.

Reply:
Hey @netroy !

I think this should now be ok. Took the opportunity to refactor a little bit and make it more clear what we are trying to do. So now we have a single class `McpServerManager` (as opposed to `McpServerSingleton` & the misnamed `McpServer`) which is responsible for creating and keeping account of MCP servers and transports. Tools &  naming of the server are set when the session is established.

I did keep the singleton, but I think it's now more clear what the intention is and how to use it.